File: sdkwork-backend-sdk-flutter/lib/src/api/net.dart
import '../http/client.dart';
import '../models.dart';

class NetApi {
  final HttpClient _client;
  
  NetApi(this._client);

  /// 更新域名
  Future<PlusApiResultPlusHostDomainVO?> update(PlusHostDomainForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/net/host/domain'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusHostDomainVO ? response : null;
  }

  /// 创建域名
  Future<PlusApiResultPlusHostDomainVO?> create(PlusHostDomainForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/net/host/domain'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusHostDomainVO ? response : null;
  }

  /// Update an existing DNS record
  Future<PlusApiResultPlusDnsRecordVO?> updateRecord(PlusDnsRecordForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/net/dns/record'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusDnsRecordVO ? response : null;
  }

  /// Create a new DNS record
  Future<PlusApiResultPlusDnsRecordVO?> createRecord(PlusDnsRecordForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/net/dns/record'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusDnsRecordVO ? response : null;
  }

  /// 分页获取域名
  Future<PlusApiResultPagePlusHostDomainVO?> createListByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/net/host/domain/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusHostDomainVO ? response : null;
  }

  /// 获取所有域名
  Future<PlusApiResultListPlusHostDomainVO?> createListAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/net/host/domain/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusHostDomainVO ? response : null;
  }

  /// Get DNS records by page
  Future<PlusApiResultPagePlusDnsRecordVO?> createListByPageRecord(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/net/dns/record/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusDnsRecordVO ? response : null;
  }

  /// Get all DNS records
  Future<PlusApiResultListPlusDnsRecordVO?> createListAllEntitiesRecord(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/net/dns/record/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusDnsRecordVO ? response : null;
  }

  /// 获取域名详情
  Future<PlusApiResultPlusHostDomainVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/net/host/domain/${id}'));
    return response is PlusApiResultPlusHostDomainVO ? response : null;
  }

  /// 删除域名
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/net/host/domain/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }

  /// Get a DNS record by ID
  Future<PlusApiResultPlusDnsRecordVO?> getByIdRecord(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/net/dns/record/${id}'));
    return response is PlusApiResultPlusDnsRecordVO ? response : null;
  }

  /// Delete a DNS record
  Future<PlusApiResultBoolean?> deleteRecord(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/net/dns/record/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/net.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusDnsRecordVO, PlusApiResultListPlusHostDomainVO, PlusApiResultPagePlusDnsRecordVO, PlusApiResultPagePlusHostDomainVO, PlusApiResultPlusDnsRecordVO, PlusApiResultPlusHostDomainVO, PlusDnsRecordForm, PlusHostDomainForm, QueryListForm

class NetApi:
    """net API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusHostDomainForm) -> PlusApiResultPlusHostDomainVO:
        """更新域名"""
        return self._client.put(f"/backend/v3/api/net/host/domain", json=body)

    def create(self, body: PlusHostDomainForm) -> PlusApiResultPlusHostDomainVO:
        """创建域名"""
        return self._client.post(f"/backend/v3/api/net/host/domain", json=body)

    def update_record(self, body: PlusDnsRecordForm) -> PlusApiResultPlusDnsRecordVO:
        """Update an existing DNS record"""
        return self._client.put(f"/backend/v3/api/net/dns/record", json=body)

    def create_record(self, body: PlusDnsRecordForm) -> PlusApiResultPlusDnsRecordVO:
        """Create a new DNS record"""
        return self._client.post(f"/backend/v3/api/net/dns/record", json=body)

    def create_list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusHostDomainVO:
        """分页获取域名"""
        return self._client.post(f"/backend/v3/api/net/host/domain/list", json=body, params=params)

    def create_list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusHostDomainVO:
        """获取所有域名"""
        return self._client.post(f"/backend/v3/api/net/host/domain/list/all", json=body)

    def create_list_by_page_record(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusDnsRecordVO:
        """Get DNS records by page"""
        return self._client.post(f"/backend/v3/api/net/dns/record/list", json=body, params=params)

    def create_list_all_entities_record(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusDnsRecordVO:
        """Get all DNS records"""
        return self._client.post(f"/backend/v3/api/net/dns/record/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusHostDomainVO:
        """获取域名详情"""
        return self._client.get(f"/backend/v3/api/net/host/domain/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """删除域名"""
        return self._client.delete(f"/backend/v3/api/net/host/domain/{id}")

    def get_by_id_record(self, id: str) -> PlusApiResultPlusDnsRecordVO:
        """Get a DNS record by ID"""
        return self._client.get(f"/backend/v3/api/net/dns/record/{id}")

    def delete_record(self, id: str) -> PlusApiResultBoolean:
        """Delete a DNS record"""
        return self._client.delete(f"/backend/v3/api/net/dns/record/{id}")
